common/src/main/java/com/wynntils/features/user/QuestBookFeature.java

```diff
@@ -0,0 +1,134 @@
+import com.wynntils.core.keybinds.KeyBind;
+import com.wynntils.core.managers.ContainerQueryManager;
+import com.wynntils.wynn.event.WorldStateEvent;
+import com.wynntils.wynn.model.WorldStateManager;
+import com.wynntils.wynn.utils.InventoryUtils;
+import java.util.List;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import org.lwjgl.glfw.GLFW;
+
+public class QuestBookFeature extends UserFeature {
+    private static final String QUESTS_TITLE = "§0[Pg. 1] §8mag_icus'§0 Quests";
+    private static final String QUESTS_TITLE_2 = "§0[Pg. 2] §8mag_icus'§0 Quests";
+    private static final String QUESTS_TITLE_3 = "§0[Pg. 3] §8mag_icus'§0 Quests";
+
+    @RegisterKeyBind
+    private final KeyBind questBookKeyBind =
+            new KeyBind("Read Quest Book", GLFW.GLFW_KEY_F, true, this::onScanQuestBook);
+
+    private void onScanQuestBook() {
+        ContainerQueryManager.start((ignored, title, menuType) -> {
+            ContainerQueryManager.openInventory(InventoryUtils.QUEST_BOOK_SLOT_NUM);
+            ContainerQueryManager.nextAction = (itemsPg1, titlePg1, menuTypePg1) -> {
+                System.out.println("GOT PAGE 1:" + titlePg1.getString() + ": " + itemsPg1);
+                if (titlePg1.getString().equals(QUESTS_TITLE)) {
+                    System.out.println("title ok");
+                } else {
+                    System.out.println("title not ok");
+                }
+                if (isPage2Button(itemsPg1)) {
+                    ContainerQueryManager.clickOnSlot(itemsPg1, 8);
+                }
+                ContainerQueryManager.nextAction = (itemsPg2, titlePg2, menuTypePg2) -> {
+                    System.out.println("GOT PAGE 2:" + titlePg2.getString() + ": " + itemsPg2);
+                    if (titlePg2.getString().equals(QUESTS_TITLE_2)) {
+                        System.out.println("title ok");
+                    } else {
+                        System.out.println("title not ok");
+                    }
+                    if (isPage3Button(itemsPg2)) {
+                        ContainerQueryManager.clickOnSlot(itemsPg2, 8);
+                    }
+                    ContainerQueryManager.nextAction = (itemsPg3, titlePg3, menuTypePg3) -> {
+                        System.out.println("GOT PAGE 3:" + titlePg3.getString() + ": " + itemsPg3);
+                        if (titlePg3.getString().equals(QUESTS_TITLE_3)) {
+                            System.out.println("title ok");
+                        } else {
+                            System.out.println("title not ok");
+                        }
+                        return null;
+                    };
+
+                    return QUESTS_TITLE_3;
+                };
+                return QUESTS_TITLE_2;
+            };
+            return QUESTS_TITLE;
+        });
+    }
+
+    private void onScanQuestBook2() {
```

Comment:
@kristofbolyai Does this seem like a good API to "script" an interaction with a container?

Reply:
Seems fine, although it is harder to debug for sure.

Reply:
When you say "harder to debug", you're not comparing with [QuestManager.readQuestBook(()](https://github.com/Wynntils/Wynntils/blob/fdde2d1fece9f419615a72fd36b0b75415684eec/src/main/java/com/wynntils/modules/questbook/managers/QuestManager.java#L83) in Legacy, are you? :)

Reply:
But I am open for suggestions on how to improve debuggability. My goal is to have a way to easily describe an interaction with a "hidden" inventory, where you can click on stuff, check the results and build up some kind of model of what you can extract from the information.

Reply:
First off, what do you think of having a "QueryBuilder" class, then have a build method, to get the actual query you execute. That means good code separation, and you can step through code easier, as the containers used to build the query are not mixed up with the executing query's container.

Reply:
That was kind of what I was aiming at, but it might not have been completely separated. Let me update my sketch.

Reply:
@kristofbolyai Something like this? Is that more of what you where thinking of?

Reply:
Seems better, but we will need to see the actual impl. Structural changes wouldn't require logic rewrite.